build-tools/packages/version-tools/src/test/internalVersionScheme.test.ts

```diff
@@ -106,6 +130,25 @@ describe("internalScheme", () => {
 			assert.isTrue(isInternalVersionRange(input));
 		});
 
+		it(">=2.0.0-rc.1.0.0 <2.0.0-rc.1.1.0 is internal", () => {
+			const input = `>=2.0.0-rc.1.0.0 <2.0.0-rc.1.1.0`;
+			assert.isTrue(isInternalVersionRange(input));
+		});
+
+		// This test case should fail but it doesn't. It's skipped because I think the case it guards against isn't likely,
+		// so I don't think it's worth the cost of fixing it.
+		//
+		// The reason the code behave wrong is because it only checks the lower bound of the range. That's all the semver
+		// library provides - semver.minVersion(range). Since ranges need not hane an upper bound, this is an undertsnadable
+		// omission, but it means that we're on our own to parse the range and get an upper bound version from it. Once we
+		// have the two versions then we could parse them and compare their prerelease identifiers to make sure they match.
+		//
+		// But that is quite a lot of code to add and maintain for what should be a rare confition.
+		it(">=2.0.0-internal.1.0.0 <2.0.0-rc.1.1.0 is not internal", () => {
+			const input = `>=2.0.0-internal.1.0.0 <2.0.0-rc.1.1.0`;
+			assert.isFalse(isInternalVersionRange(input));
+		}).skip();
+
```

Comment:
I am not sure that I agree the result is incorrect.
I would have thought anything that allows "internal" (really now official prerelease) is "internal".
And "*-rc.*" is also "internal"; so why wouldn't this be "internal"?

Reply:
I'm trying to keep the definition of a "Fluid internal version range" as narrow as possible. Unless we have a need for
this (valid) semver range to be considered a valid Fluid range, I'd prefer to say that valid ranges always use the same
prerelease ID.

Reply:
So, "only" internal ranges. Then I think you want `subset` with the super set being `>=${min} <=${maxMinPrerelease}`
where min = `minVersion(range)` and maxMinPrerelease = `${<public of min>}-${<prerelease tag of min>}.x`. If that fails, then you know you have more than one range. I think you want to test each `set` of `Range` (`||` ranges) for that. Probably you already should be checking each `set`.

Reply:
I am guessing the superset range could actually just be `${<public of min>}-${<prerelease tag of min>}.x`.

Reply:
No opposition to leaving fix for a later time. I would update the comment though to note things more explicitly. There are typos in the comment; so, replacement would address that too. ;)
I looked at `set` more closely. It looks like you need to do a join(' ') on each set entry (array of comparators) to get a range that you can check. (They should have called it `sets`.)

Reply:
I'm going to leave this as-is for now, since the bulk of this change is needed to enable the pipelines to work with the rc version.

Reply:
I wanted to see if I was talking junk and I was a little bit: #18828